### PR TITLE
Adds a jobban checking verb. 🔨

### PR DESCRIPTION
**Pull request by FlattestGuitar**
I can't get XAMPP to work over here because of my illiteracy, so this is completely untested.

I'd appreciate if someone took the code and tested it themselves if they have a mySQL set up on their local server.

:cl: FlattestGuitar
rscadd: Adds a jobban-checking verb to the OOC tab.
/:cl:

### CODE REVIEW

**Comment by Crazylemon64** [comment]
No message is being printed when I use the verb

**Comment by FlattestGuitar** [comment]
Not good at all!

I'm having a problem as well, but I'm only checking the legacy system. Somehow `if(!ckey in jobban_keylist)` fails, so it doesn't tell you you have no bans, but then `for(var/J in jobban_assoclist[ckey])` gives out nothing at all, still not printing anything.

I'm confused.

**Comment by FlattestGuitar** [comment]
Yeah, I tried, I failed. I can't figure out how to get a DB working on my end; I'll leave this to someone who knows what's what.

**Comment by Kyep** [comment]
I'll help you get your DB working, just ping me sometime you see me in-game.

**Comment by FlattestGuitar** [comment]
After thorough testing, I still can't get the legacy system to act consistently, but with it works just fine with SQL bans.

I'd really appreciate someone looking into what might be wrong here. I'll just add an alert that the legacy system is incompatible otherwise - no proper server should be using it anyway.

**Comment by tigercat2000** [comment]
@FlattestGuitar The legacy system outright does not support jobbans, only temp and permabans. Don't worry about it. We might as well rip it out entirely.

**Comment by FlattestGuitar** [comment]
I killed the legacy ban support for the verb.

rip

File: code/modules/admin/banjob.dm [hunk 1]
```diff
@@ -140,3 +140,58 @@ DEBUG
...
+			var/reason = select_query.item[3]
+			var/job = select_query.item[4]
+			var/duration = select_query.item[5]
+			var/expiration = select_query.item[6]
+			var/ackey = select_query.item[7]
+
+			if(bantype != null)
+				is_actually_banned = 1
+
+			to_chat(src, "<span class='warning'>[bantype]: [job] - REASON: [reason], by [ackey]; [bantime]; [duration]; expires [expiration]</span>")
```
Comment (Kyep) [inline code/modules/admin/banjob.dm]: Should this be under the if(bantype != null) conditional?

File: code/modules/admin/banjob.dm [hunk 2]
```diff
@@ -140,3 +140,58 @@ DEBUG
...
+			var/bantime = select_query.item[5]
+
+			if(bantype != null)
+				is_actually_banned = 1
+
+			var/output = "[bantype]: [bantime] [job] - REASON: [reason], by [ackey]"
+			to_chat(src, "<span class='warning'>[output]</span>")
+
+		//temps
+		select_query = dbcon.NewQuery("SELECT bantime, bantype, reason, job, duration, expiration_time, a_ckey FROM [format_table_name("ban")] WHERE ckey like '[ckey]' AND bantype like 'JOB_TEMPBAN' AND expiration_time > Now() AND isnull(unbanned) ORDER BY bantime DESC LIMIT 100")
```
Comment (Kyep) [inline code/modules/admin/banjob.dm]: Why do two different SQL queries for job tempbans, v job permabans?

File: code/modules/admin/banjob.dm (hunk 1 above)
Comment (Kyep) [inline code/modules/admin/banjob.dm]: Also, this doesn't provide any link to appeal the bans.

File: code/modules/admin/banjob.dm [hunk 3]
```diff
@@ -140,3 +140,58 @@ DEBUG
 				jobban_savebanfile()
 			return 1
 	return 0
+
+/mob/verb/displayjobbans()
```
Comment (Kyep) [inline code/modules/admin/banjob.dm]: Perhaps include a clickable link to this on the job preferences screen, if the user has active job bans?

File: code/modules/admin/banjob.dm [hunk 4]
```diff
@@ -140,3 +140,58 @@ DEBUG
...
+
+			var/bantime = select_query.item[1]
+			var/bantype  = select_query.item[2]
+			var/reason = select_query.item[3]
+			var/job = select_query.item[4]
+			var/duration = select_query.item[5]
+			var/expiration = select_query.item[6]
+			var/ackey = select_query.item[7]
+
+			if(bantype != null)
```
Comment (Kyep) [inline code/modules/admin/banjob.dm]: Why not put this in the SQL query?

File: code/modules/admin/banjob.dm (hunk 3 above)
Comment (FlattestGuitar) [inline code/modules/admin/banjob.dm]: Seems like slight overkill to me, plus I'd rather keep nanoUI out of this PR for the sake of simplicity.

Another day.

**Comment by FlattestGuitar** [comment]
Addressed all of Kyet's comments.

**Comment by Crazylemon64** [comment]
Cool, it works

**Pull request merged**